tests/utils.py

```
@@ -18,3 +18,30 @@ def debug_json(json_data: JSONValue) -> None:  # pragma: no cover
+    def _convert_int_keys(self, obj: Any) -> Any:  # pyright: ignore [reportAny, reportExplicitAny]
+        """Recursively convert string keys that represent positive integers to int."""
+        if isinstance(obj, dict):
+            return {
+                self._convert_key(k): self._convert_int_keys(v) for k, v in obj.items()
+            }
+        if isinstance(obj, list):
+            return [self._convert_int_keys(item) for item in obj]
+        return obj  # pyright: ignore [reportAny]
+
+    @staticmethod
+    def _convert_key(key: Any) -> int | Any:  # pyright: ignore [reportAny, reportExplicitAny]
```

Ruff (0.8.2)

Dynamically typed expressions (typing.Any) are disallowed in `key`

(ANN401)